src/yunohost/app.py

```diff
@@ -2737,46 +2856,76 @@ def parse_question(self, question, user_answers):
     def parse(self, question, user_answers):
         question = self.parse_question(question, user_answers)
 
-        if question.value is None:
-            text_for_user_input_in_cli = self._format_text_for_user_input_in_cli(
-                question
-            )
-
-            try:
-                question.value = Moulinette.prompt(
-                    text_for_user_input_in_cli, self.hide_user_input_in_prompt
+        while True:
+            # Display question if no value filled or if it's a readonly message
+            if Moulinette.interface.type== 'cli':
+                text_for_user_input_in_cli = self._format_text_for_user_input_in_cli(
+                    question
                 )
-            except NotImplementedError:
-                question.value = None
+                if getattr(self, "readonly", False):
+                    Moulinette.display(text_for_user_input_in_cli)
+
+                elif question.value is None:
+                    prefill = ""
+                    if question.current_value is not None:
+                        prefill = question.current_value
+                    elif question.default is not None:
+                        prefill = question.default
+                    question.value = Moulinette.prompt(
+                        message=text_for_user_input_in_cli,
+                        is_password=self.hide_user_input_in_prompt,
+                        confirm=self.hide_user_input_in_prompt,
+                        prefill=prefill,
+                        is_multiline=(question.type == "text")
+                    )
 
-        # we don't have an answer, check optional and default_value
-        if question.value is None or question.value == "":
-            if not question.optional and question.default is None:
-                raise YunohostValidationError(
-                    "app_argument_required", name=question.name
-                )
-            else:
+
+            # Apply default value
+            if question.value in [None, ""] and question.default is not None:
```

Comment:
With boolean or integer value False or 0 means something. Don't know if at this step it could be of that types...
Maybe we should check [] to for Tags args